### ISSUE TITLE
ModuleNotFoundError: No module named 'six.moves'

### ISSUE THREAD

**Issue by hugoboss15** [benjaminp/six#384]
There is no moves.py, no __init__.py file, no six main package folder. six.py can be found in site-packages. six.moves has to be implemented in six.py. Unfortunately I get the Error ModuleNotFoundError: No module named 'six.moves' even after uninstalling and installing again. I run Python 3.12.3. and six1.12.0 on macOS.

**Comment by hugoboss15** [comment]
Update 20.05.24 18:58 GMT+2: After using "pip3 uninstall textract" for uninstalling textract and unsing "pip3 install --upgrade six" for upgrading to six 1.16.0 Jupyter notebook runs.

**Comment by apitofme** [comment]
I'm not sure how [this issue](https://github.com/Xirider/LiveCode/issues/18#issuecomment-2139230202) relates but I've had the same error come up whilst using the **LiveCode** extension for VSCode:
![Screenshot from 2024-05-30 12-27-40](https://github.com/benjaminp/six/assets/1789853/3ffb4338-5e06-498a-8326-e669752781ed)

Which I initially thought was cause by the [astunparse](https://github.com/simonpercivall/astunparse) module, given the error output:
```
Traceback (most recent call last):
File "c:\Users\leandre.vscode\extensions\xirider.livecode-1.3.10\node_modules\arepl-backend\python\arepl_python_evaluator.py", line 11, in
import astunparse
File "c:\Users\leandre.vscode\extensions\xirider.livecode-1.3.10\node_modules\arepl-backend\python\astunparse_init_.py", line 3, in
from six.moves import cStringIO
ModuleNotFoundError: No module named 'six.moves'
```

However (given this thread) it appears that this may not be the case, so I don't know what the problem is with **six** that is causing the 'missing' _moves_ module in this situation.

All I can say is that editing `astunparse.__init__.py` to change the import line from this `from six.moves import cStringIO` to this `from io import StringIO` and updating the references seems to fix the immediate problem running the **LiveCode** extension.
<br />

Basically what I'm asking is:
-- Is this an issue with **six** itself, or is it an issue with how these other libraries / applications are using it?

**Comment by hugoboss15** [comment]
Thanks for the reply! Since my last comment "Update 20.05.24 18:58 GMT+2: After using "pip3 uninstall textract" for uninstalling textract and using "pip3 install --upgrade six" for upgrading to six 1.16.0 Jupyter notebook runs." this prooblem is resolved.

**Comment by apitofme** [comment]
@hugoboss15 That may be the case for you but other people may still be having an issue related to this error, and whilst I found a patch to fix my immediate problem I was reporting here to try to ascertain if this is part of a larger problem or if anyone here can shed light on why this error happened and where it needs to be addressed -- Thanks

**Comment by hugoboss15** [comment]
@apitofme When using the version six1.12.0, the six.moves module did not launch. six1.16.0 was automatically uninstalled right after I installed textract1.6.5 library. I had to manually uninstall textract1.6.5 library and upgrade to the most recent six1.16.0 version to use it with six.moves. The optimal situationI would be being able to use six1.16.0 with six.moves and textract1.6.5 library. I hope this helps other people to understand this error better, and to partially solve it.

**Comment by apitofme** [comment]
Thanks @hugoboss15 but I'm not sure how that is relevant to my issue, unless `python-textract` is some kind of hidden dependency to the `python-six` module?

If it is helps I can confirm that I do **not** have the _textract_ library installed on my system and (to the best of my knowledge) never have.

The title to this thread is:

> ModuleNotFoundError: No module named 'six.moves'

My error code stated:

> ModuleNotFoundError: No module named 'six.moves'

...ergo I felt this was the right place to post my issue, rather than start a new thread?!

Could this not be a case of: Same error -- Different cause ?

I was just posting in case a maintainer or contributor could determine whether this is or isn't an issue with this module, or if the issue is being caused elsewhere (e.g. by reference in an external module due to incorrect usage, deprecated features or version mismatch).

**Comment by hugoboss15** [comment]
I actually do not know if six1.16.0 or six1.12.0 is even necessary for running textract1.6.5 on Python 3.12.3. At least six1.12.0 was automatically installed, when I installed textract1.6.5, that's the dependency. After Upgrading to six1.16.0, six.moves stopped working, if I recall this correctly.

**Comment by anurag-ctds** [comment]
```
(venv) Cognitrees-MacBook-Air:ccm anuragm$ ccm node1 cqlsh
/Users/anuragm/Desktop/ccm/venv/bin/ccm:7: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
  import pkg_resources
Traceback (most recent call last):
  File "/Users/anuragm/.ccm/repository/4.1.3/bin/cqlsh.py", line 134, in <module>
    from cassandra.cluster import Cluster
  File "/Users/anuragm/.ccm/repository/4.1.3/bin/../lib/cassandra-driver-internal-only-3.25.0.zip/cassandra-driver-3.25.0/cassandra/cluster.py", line 33, in <module>
ModuleNotFoundError: No module named 'six.moves'
```

I am also getting the same ModuleNotFound error, I tried few things like reinstalling, upgrading and what not. But nothing seems to work. Does anyone have any idea how can it be fixed? - Thanks

**Comment by Chharlie1130** [comment]
> I'm not sure how [this issue](https://github.com/Xirider/LiveCode/issues/18#issuecomment-2139230202) relates but I've had the same error come up whilst using the **LiveCode** extension for VSCode: ![Screenshot from 2024-05-30 12-27-40](https://private-user-images.githubusercontent.com/1789853/335162641-3ffb4338-5e06-498a-8326-e669752781ed.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg2NjYwMzYsIm5iZiI6MTcyODY2NTczNiwicGF0aCI6Ii8xNzg5ODUzLzMzNTE2MjY0MS0zZmZiNDMzOC01ZTA2LTQ5OGEtODMyNi1lNjY5NzUyNzgxZWQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTAxMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEwMTFUMTY1NTM2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YTgwZmNhMDM2Yjc2Nzc3MmRlNGI4MzI0YzQ4NTcwNGRjNmNhNDU3MWFhMTA1YzY3MTJkZTYzMGMxNzRmZmFkMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.TQSrWTWpheEVWHjT4I15B9vwwrO3Pw2dHCBviEHj8F8)
> 
> Which I initially thought was cause by the [astunparse](https://github.com/simonpercivall/astunparse) module, given the error output:
> 
> ```
> Traceback (most recent call last):
> File "c:\Users\leandre.vscode\extensions\xirider.livecode-1.3.10\node_modules\arepl-backend\python\arepl_python_evaluator.py", line 11, in
> import astunparse
> File "c:\Users\leandre.vscode\extensions\xirider.livecode-1.3.10\node_modules\arepl-backend\python\astunparse_init_.py", line 3, in
> from six.moves import cStringIO
> ModuleNotFoundError: No module named 'six.moves'
> ```
> 
> However (given this thread) it appears that this may not be the case, so I don't know what the problem is with **six** that is causing the 'missing' _moves_ module in this situation.
> 
> All I can say is that editing `astunparse.__init__.py` to change the import line from this `from six.moves import cStringIO` to this `from io import StringIO` and updating the references seems to fix the immediate problem running the **LiveCode** extension.
> 
> Basically what I'm asking is: -- Is this an issue with **six** itself, or is it an issue with how these other libraries / applications are using it?

# You save my life!

**Comment by mraatek** [comment]
can someone help me with this LiveCode extension for VSCode

Print Output:
Traceback (most recent call last):
  File "c:\Users\Admin\.vscode\extensions\xirider.livecode-1.3.10\node_modules\arepl-backend\python\arepl_python_evaluator.py", line 11, in <module>
    import astunparse
  File "c:\Users\Admin\.vscode\extensions\xirider.livecode-1.3.10\node_modules\arepl-backend\python\astunparse\__init__.py", line 3, in <module>
    from six.moves import cStringIO
ModuleNotFoundError: No module named 'six.moves'

**Comment by markxma1** [comment]
same here

Print Output:
Traceback (most recent call last):
  File "/home/xma1/.vscode/extensions/xirider.livecode-1.3.10/node_modules/arepl-backend/python/arepl_python_evaluator.py", line 11, in <module>
    import astunparse
  File "/home/xma1/.vscode/extensions/xirider.livecode-1.3.10/node_modules/arepl-backend/python/astunparse/__init__.py", line 3, in <module>
    from six.moves import cStringIO
ModuleNotFoundError: No module named 'six.moves'

i have installed pip3 install --upgrade six --break-system-packages, no change. use ubuntu 24.04
